scripts/uosc.lua

```diff
@@ -907,6 +907,16 @@ function serialize_chapter_ranges(normalized_chapters)
 					end
 				end
 			end
+			-- single chapter for ad
+			if chapter.lowercase_title:find('%[sponsorblock%]:') or chapter.lowercase_title:find('^sponsors?') then
```

Comment:
This should be `elseif`.

Also, `[sponsorblock]` is just what this particular youtube creator decided to call his sponsor timestamp, so it's not going to match sponsor segment of anyone else. I think these 2 matches should be consolidated into `:find('sponsor')`, otherwise we won't match much, and might as well just not include this extra matching at all.

And how common would it be that someone puts a `sponsor` word into their chapter and it's not actually a sponsor segment?

Reply:
> This should be `elseif`.

oops, it used to be elseif, but then I tried something different and it never became elseif again.

> Also, `[sponsorblock]` is just what this particular youtube creator decided to call his sponsor timestamp, so it's not going to match sponsor segment of anyone else.

For testing I've downloaded a different video from a different creator and it still was `[sponsorblock]:`, so I think that's generally the case for downloaded videos with `yt-dlp --sponsorblock-mark "all"`.

> And how common would it be that someone puts a `sponsor` word into their chapter and it's not actually a sponsor segment?

I remember seeing some, enough to add that to my range config. I'll hunt down one for you :wink:

Reply:
Wait, the `[sponsorblock]:` thing is inserted by an alternative sponsorblock plugin built into yt-dlp?

If so, just change the `elseif`, and keep the your original 2 matches.

Reply:
The [WAN Show](https://www.youtube.com/watch?v=dOyrx5HOCyY) always has chapters with `Sponsor - Companyname`